www/docs/client/mocking.md

```diff
@@ -0,0 +1,234 @@
+) => {
+  return render(ui, {
+    wrapper: (props) => <TRPCMockedProvider {...props} />,
+    ...options,
+  });
+};
+```
+
+:::note
+
+- If you already have a similar to render a component with providers in tests, you may choose to import `TRPCMockedProvider` to that function instead, this is a simple example for demonstration purposes.
+- If you don't want to use a custom hook like the one illustrated, you may also choose to simply use the return value of the function.
+
+:::
+
+## Using mocks
+
+Now we have `TRPCMockedProvider`, we can start writing and using mocks. The following examples will use [Mock Service Worker](https://mswjs.io/) to intercept our API calls and return a response, as well as [`msw-trpc`](https://www.npmjs.com/package/msw-trpc) to utilize the type signature of your API when defining our [request handlers](https://mswjs.io/docs/getting-started/mocks/rest-api).
+
+To mock data using your own router, start by installing both packages into your dev dependendies:
+
+`npm install --save-dev msw msw-trpc`
+
+### 1. Create your MSW handlers
+
+To utilize tRPC's typing superpowers, we can create a `mswTrpc` constant which generates a typed set of handlers for MSW, based on she shape of your `AppRouter`. This enables us to build handlers specific to our individual tests rapdily and with typesafety.
+
+```ts title='trpcMockedProvider.ts'
+import type { AppRouter } from "path/to/your/router";
+import { createTRPCMsw } from "msw-trpc";
+
+// YOUR PROVIDER
+
+export const trpcMsw = createTRPCMsw<AppRouter>();
+```
+
+:::note
+
+If you want to skip this step, it is possible to create standard REST handlers using `msw`s `rest` object and defining them to your network calls.
+
+:::
+
+### 2. Use your handlers and mocks to simulate an API call
+
+Let's say that we recently defined an endpoint in our `cats` router serve everyone with information about our favorite cat.
+
+<details><summary>Backend code</summary>
+
+```ts title='server/routers/cats.ts'
+export const catsRouter = router({
+  bestCat: publicProcedure
+    .input(z.object({ name: z.string() }))
+    .query(({ input }) => {
+      return {
+        bestCat: `Hey ${input.name}! The best cat is Mozzie!`,
+      };
+    }),
+});
+```
+
+</details>
+
+In addition, we've also created a simple React component, `BestCat` that displays the return of this query.
+
+<details><summary>Frontend code</summary>
+
+```ts title='component/cats/BestCat.tsx'
+export default const BestCat = () => {
+  const { loading, error, data } = trpc.cats.bestCat.useQuery({ name: "John" })
+  if (loading) return <p>Loading...</p>;
+  return (
+    <StyledCatText>{data.bestCat}</StyledCatText>
+  )
+}
+```
+
+</details>
+
+Finally, we want to verify that our component is working as we anticipated, so we write a test.
+
+```ts title='component/cats/BestCat.spec.tsx'
+describe("BestCat", () => {
+  it("should render the result of our query", async () => {
+    render(<BestCat />);
```

Comment:
I feel like showing this incorrect example first isn't helpful. Maybe just go straight to the correct one, and explain why it's correct?